core/framework/src/main/java/org/phoebus/framework/macros/Macros.java

```diff
@@ -42,7 +42,7 @@ public class Macros implements MacroValueProvider
     private final Map<String, String> macros = new LinkedHashMap<>();
 
     /** Regular expression for valid macro names */
-    public final static Pattern MACRO_NAME_PATTERN = Pattern.compile("[A-Za-z][A-Za-z0-9_.\\-\\[\\]]*");
+    public final static Pattern MACRO_NAME_PATTERN = Pattern.compile("[A-Za-z][A-Za-z0-9_.\\@\\+%\\-\\[\\]]*");
```

Comment:
Is this necessary? We don't want to allow '@' or '+' or '%' within a macro name. Within a macro value, yes, but not a macro name.